tests/integration/tests/test_control_plane_taints.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+import time
+from typing import List
+
+import pytest
+import yaml
+from test_util import harness, util
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.mark.node_count(1)
+@pytest.mark.disable_k8s_bootstrapping()
+def test_control_plane_taints(instances: List[harness.Instance]):
+    k8s_instance = instances[0]
+
+    bootstrap_conf = yaml.safe_dump(
+        {"control-plane-taints": ["node-role.kubernetes.io/control-plane:NoSchedule"]}
+    )
+
+    k8s_instance.exec(
+        ["dd", "of=/root/config.yaml"],
+        input=str.encode(bootstrap_conf),
+    )
+
+    k8s_instance.exec(["k8s", "bootstrap", "--file", "/root/config.yaml"])
+    retries = 10
+    while retries and not (nodes := util.get_nodes(k8s_instance)):
```

Comment:
nit: Would be cool, if we could reuse the existing retry mechanisms here but I see how this would require a refactoring of the `util.get_nodes` function.

Reply:
right, i looked at this too but didn't find a good way in this PR to accomplish this.

Reply:
something for another time...